loader/linux/icd_linux.c

```diff
@@ -61,76 +62,74 @@ void khrIcdOsVendorsEnumerate(void)
         // attempt to load all files in the directory
         for (dirEntry = readdir(dir); dirEntry; dirEntry = readdir(dir) )
         {
-            switch(dirEntry->d_type)
+            struct stat statBuff;
+            stat(dirEntry->d_name, &statBuff);
```

Comment:
This PR has broken our internal tests, because this `stat` call is always failing with "No such file or directory".

I think this needs to be the full absolute path (e.g. the `fileName` string that is constructed below), instead of the relative `dirEntry->d_name` path used here.

Reply:
You are absolutely correct, I made a fix pull request here: https://github.com/KhronosGroup/OpenCL-ICD-Loader/pull/161
Thanks for finding this.